code/modules/library/random_books.dm

```diff
@@ -0,0 +1,64 @@
+/obj/item/weapon/book/manual/random/New()
+	var/static/banned_books = list(/obj/item/weapon/book/manual,/obj/item/weapon/book/manual/random,/obj/item/weapon/book/manual/nuclear,/obj/item/weapon/book/manual/wiki)
+	var/newtype = pick(typesof(/obj/item/weapon/book/manual) - banned_books)
+	new newtype(loc)
+	qdel(src)
+
+/obj/structure/bookcase/random
+	var/category = null
+	var/book_count = 2
+	anchored = 1
+	state = 2
+
+/obj/structure/bookcase/random/New()
+	if(!book_count)
+		update_icon()
+		return
+	if(!establish_db_connection())
+		if(prob(5))
+			var/obj/item/weapon/paper/P = new(get_turf(loc))
+			P.info = "There once was a book from Nantucket<br>But the database failed us, so f*$! it.<br>I tried to be good to you<br>Now this is an I.O.U<br>If you're feeling entitled, well, stuff it!<br><br><font color='gray'>~</font>"
+		update_icon()
+		return
+	if(category && prob(25)) category = null
+	book_count += pick(-1,-1,0,1,1)
+	if(book_count <= 0)
+		update_icon()
+		return
+
+	var/static/l = format_table_name("library")
+	var/c = category? "AND category='[category]'" :""
+	var/q = "SELECT * FROM [l] WHERE isnull(deleted) [c] GROUP BY title ORDER BY rand() LIMIT [book_count];"
+	var/DBQuery/query = dbcon.NewQuery(q)
+	query.Execute()

```

Comment:
Just in case, could you add an error report to this execute call; e.g. https://github.com/tgstation/-tg-station/blob/master/code/modules/admin/sql_notes.dm#L12